base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/ScriptGeneratorSingleton.java

```diff
@@ -576,6 +589,33 @@ public void refreshParameterValidityChecking() throws NoScriptDefinitionSelected
 			threadError = true;
 		}
 	}
+	
+    /**
+     * Refresh the time estimate of the parameters, or if it fails refresh the
+     * error state of the model to be listened to by the ViewModel.
+     * 
+     * @throws NoScriptDefinitionSelectedException If there is no script definition
+     *                                             selected to refresh checking
+     *                                             against.
+     */
+    public void refreshTimeEstimation() throws NoScriptDefinitionSelectedException {
+        ScriptDefinitionWrapper scriptDefinition = getScriptDefinition()
+                .orElseThrow(() -> new NoScriptDefinitionSelectedException(
+                        "Tried to refresh time estimation with no script definition selected"));
+        try {
+            generator.refreshTimeEstimation(scriptGeneratorTable, scriptDefinition);
+            languageSupported = true;
+            threadError = false;
+        } catch (UnsupportedLanguageException e) {
+            firePropertyChange(LANGUAGE_SUPPORT_PROPERTY, languageSupported, false);
+            LOG.error(e);
+            languageSupported = false;
```

Comment:
```suggestion
            firePropertyChange(LANGUAGE_SUPPORT_PROPERTY, languageSupported, languageSupported=false);
            LOG.error(e);
```
Similar for thread error below.

Reply:
This same pattern of error handling is used in lots of places in this class, can we generalise it?